core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -92,12 +94,22 @@ default ViewVersion currentVersion() {
 
   @Value.Derived
   default Map<Integer, ViewVersion> versionsById() {
-    return Builder.indexVersions(versions());
+    ImmutableMap.Builder<Integer, ViewVersion> builder = ImmutableMap.builder();
```

Comment:
No need for separate static methods now that these are used directly by the builder in its constructor.